### PR TITLE
Add `JSScheduler` and `JSPromise` publisher

### PR DESCRIPTION
**Pull request by MaxDesiatov**
`JSScheduler` is copied from https://github.com/TokamakUI/Tokamak/pull/281.

You can see the `JSPromise` publisher in action in combination with `fetch` by running `carton dev` in the root directory. I saw some issues with `JSValueDecoder`, which is why it's not used in this PR. I will investigate `JSValueDecoder` issues separately.

### CODE REVIEW

File: Sources/OpenCombineJS/JSPromise.swift [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    private var parent: JSPromise?
+
+    /// Reference to a `then` success callback promise instance to prevent early deallocation
+    private var then: JSPromise<JSValue, Failure>?
+
+    /// `Future` instance that handles subscriptions to this publisher.
+    private var future: Future<Success, Failure>?
+
+    fileprivate init(parent: JSPromise) {
+      future = .init { resolver in
```
Comment (kateinoigakukun) [inline Sources/OpenCombineJS/JSPromise.swift]:
```suggestion
      future = .init { [weak self] resolver in
```

Need to capture `self` weakly here

**Review by kateinoigakukun — CHANGES_REQUESTED** [review]

**Comment by kateinoigakukun** [comment]
BTW, building OpenCombine on Xcode is failing. Could you take a look on this?

File: Sources/OpenCombineJS/JSPromise.swift (hunk 1 above)
Comment (MaxDesiatov) [reply Sources/OpenCombineJS/JSPromise.swift]: I've tried that, but it didn't make any difference, `Future` doesn't capture this closure at all and executes it immediately. It shouldn't be marked `@escaping` in the first place, but was marked so for compatibility with Apple's Combine I assume.

**Comment by MaxDesiatov** [comment]
Yes, I'll probably start submitting my changes upstream, which is guaranteed to work on non-WASI platforms. I just need to update `swiftwasm-action` to use the latest toolchain and ship `carton` so that we can run `carton test` with upstream OpenCombine CI.

File: Sources/OpenCombineJS/JSPromise.swift (hunk 1 above)
Comment (kateinoigakukun) [reply Sources/OpenCombineJS/JSPromise.swift]: That depends on their implementation and capture behavior can be changed. So I think it's better to capture self weakly.

Comment (MaxDesiatov) [reply Sources/OpenCombineJS/JSPromise.swift]: Makes sense, that's updated now 🙂

**Review by kateinoigakukun — APPROVED** [review]

**Pull request merged**